File: src/R/systems/validate_phase13c_freeze.R
#!/usr/bin/env Rscript
# Independent base-R freeze-boundary validation for accepted Phase 13C.
args <- commandArgs(trailingOnly=TRUE)
root <- if (length(args)) normalizePath(args[[1]], mustWork=TRUE) else normalizePath(".", mustWork=TRUE)
reports <- file.path(root, "reports")
source_commit <- "c746e361e052e979227e40140c6d923557911524"
final_name <- "phase13c_infectious_disease_futures_freeze_manifest.json"
working_name <- "infectious_disease_scenario_manifest.json"
phase13a_name <- "phase13a_infectious_disease_freeze_manifest.json"
phase13b_name <- "phase13b_infectious_disease_dependencies_freeze_manifest.json"
text_ext <- c("csv", "json", "md", "txt", "yml", "yaml", "svg", "py", "r", "R")
expected_final <- c(
  "data/processed/scenarios/infectious_disease_scenario_assumptions.csv",
  "data/processed/scenarios/infectious_disease_transmission_states_scenario.csv",
  "data/processed/scenarios/infectious_disease_surveillance_states_scenario.csv",
  "data/processed/scenarios/infectious_disease_response_states_scenario.csv",
  "data/processed/scenarios/infectious_disease_dependency_states_scenario.csv",
  "data/processed/scenarios/infectious_disease_uncertainty_states_scenario.csv",
  "data/processed/analysis/infectious_disease_scenario_sources.csv",
  "outputs/figures/infectious_disease_scenario_comparison.csv",
  "outputs/maps/systems/43_infectious_disease_system_futures_2050.png",
  "outputs/maps/systems/43_infectious_disease_system_futures_2050.svg",
  "outputs/maps/systems/43b_infectious_disease_system_futures_2075.png",
  "outputs/maps/systems/43b_infectious_disease_system_futures_2075.svg",
  "outputs/figures/infectious_disease_scenario_comparison.png",
  "outputs/figures/infectious_disease_scenario_comparison.svg",
  "reports/infectious_disease_scenario_sources.md",
  "reports/infectious_disease_scenario_assumptions.md",
  "reports/infectious_disease_scenario_findings.md",
  "reports/infectious_disease_scenario_qa.md",
  "reports/phase13c_citation_ledger.json",
  "src/python/systems/build_infectious_disease_futures.py",
  "src/python/systems/validate_infectious_disease_futures.py",
  "src/R/systems/validate_infectious_disease_futures.R",
  "reports/infectious_disease_scenario_artifact_check.json",
  "reports/infectious_disease_scenario_independent_review.md",
  "reports/infectious_disease_scenario_manifest.json",
  "docs/phase_briefs/phase13c_infectious_disease_futures.md",
  "src/python/systems/validate_phase13c_freeze.py",
  "src/R/systems/validate_phase13c_freeze.R"
)
expected_working <- expected_final[1:24]
expected_counts <- c(scenario_assumptions=36L, transmission_states=36L, surveillance_states=36L, response_states=36L, dependency_states=36L, uncertainty_states=36L, scenario_sources=33L, comparison_rows=6L)
stop_if <- function(condition, message) if (!isTRUE(condition)) stop(message, call.=FALSE)
json_text <- function(path) paste(readLines(path, warn=FALSE, encoding="UTF-8"), collapse=" ")
raw_bytes <- function(path) readBin(path, "raw", n=as.numeric(file.info(path)$size))
sha256_raw <- function(raw, suffix=".bin") {
  tmp <- tempfile(fileext=suffix); on.exit(unlink(tmp), add=TRUE); writeBin(raw, tmp)
  cmd <- Sys.which("sha256sum")
  if (nzchar(cmd)) {
    out <- system2(cmd, tmp, stdout=TRUE, stderr=TRUE); text <- paste(out, collapse=" ")
    hits <- regmatches(text, gregexpr("[0-9a-fA-F]{64}", text, perl=TRUE))[[1]]
    stop_if(length(hits) >= 1L, "sha256sum failed")
    return(tolower(hits[[1]]))
  }
  certutil <- Sys.which("certutil"); stop_if(nzchar(certutil), "Neither sha256sum nor certutil is available")
  out <- system2(certutil, c("-hashfile", tmp, "SHA256"), stdout=TRUE, stderr=TRUE)
  hits <- tolower(gsub("[[:space:]]+", "", out)); hits <- hits[grepl("^[0-9a-f]{64}$", hits)]
  stop_if(length(hits) >= 1L, "certutil failed"); hits[[1]]
}
portable_match <- function(path, expected, recorded=NA_real_) {
  stop_if(file.exists(path) && file.info(path)$size > 0, paste("missing artifact", path))
  raw <- raw_bytes(path); hashes <- c(sha256_raw(raw, tools::file_ext(path))); lengths <- c(length(raw))
  ext <- tools::file_ext(path)
  if (tolower(ext) %in% tolower(text_ext)) {
    canonical <- charToRaw(gsub("\r\n?", "\n", rawToChar(raw), perl=TRUE))
    crlf <- charToRaw(gsub("\n", "\r\n", rawToChar(canonical), fixed=TRUE))
    hashes <- c(hashes, sha256_raw(canonical, ext), sha256_raw(crlf, ext)); lengths <- c(lengths, length(canonical), length(crlf))
  }
  tolower(expected) %in% hashes && (is.na(recorded) || as.numeric(recorded) %in% lengths)
}
manifest_entries <- function(path) {
  lines <- readLines(path, warn=FALSE, encoding="UTF-8")
  collection <- if (any(grepl('"artifacts"[[:space:]]*:[[:space:]]*\\{', lines, perl=TRUE))) "artifacts" else "files"
  start <- which(grepl(paste0('"', collection, '"'), lines, fixed=TRUE) & grepl("{", lines, fixed=TRUE))[1]
  stop_if(!is.na(start), paste("missing manifest collection", path))
  rel <- character(); hashes <- character(); bytes <- numeric(); current <- NA_character_; current_hash <- NA_character_; current_bytes <- NA_real_
  append_current <- function() {
    if (!is.na(current) && !is.na(current_hash)) { rel <<- c(rel, current); hashes <<- c(hashes, current_hash); bytes <<- c(bytes, current_bytes) }
    current <<- NA_character_; current_hash <<- NA_character_; current_bytes <<- NA_real_
  }
  if (start + 1L <= length(lines)) for (line in lines[(start + 1L):length(lines)]) {
    if (grepl("^  \\},?[[:space:]]*$", line)) { append_current(); break }
    direct <- regexec('^[[:space:]]*"([^"]+/[^"]+)"[[:space:]]*:[[:space:]]*"([0-9a-f]{64})"', line, perl=TRUE); hit <- regmatches(line, direct)[[1]]
    if (length(hit)==3L) { append_current(); rel <- c(rel, hit[[2]]); hashes <- c(hashes, hit[[3]]); bytes <- c(bytes, NA_real_); next }
    nested <- regexec('^[[:space:]]*"([^"]+/[^"]+)"[[:space:]]*:[[:space:]]*\\{', line, perl=TRUE); hit <- regmatches(line, nested)[[1]]
    if (length(hit)==2L) { append_current(); current <- hit[[2]] }
    h <- regexec('"sha256"[[:space:]]*:[[:space:]]*"([0-9a-f]{64})"', line, perl=TRUE); hh <- regmatches(line, h)[[1]]
    if (length(hh)==2L && !is.na(current)) current_hash <- hh[[2]]
    b <- regexec('"bytes"[[:space:]]*:[[:space:]]*([0-9]+)', line, perl=TRUE); bb <- regmatches(line, b)[[1]]
    if (length(bb)==2L && !is.na(current)) current_bytes <- as.numeric(bb[[2]])
  }
  append_current(); data.frame(rel=rel, sha256=hashes, bytes=bytes, stringsAsFactors=FALSE)
}
verify_manifest <- function(name, expected_n, expected_rels=character()) {
  path <- file.path(reports, name); stop_if(file.exists(path), paste("missing manifest", name)); x <- manifest_entries(path)
  stop_if(nrow(x)==expected_n, paste(name, "artifact count", nrow(x)))
  if (length(expected_rels)) stop_if(identical(as.character(x$rel), expected_rels), paste(name, "artifact inventory"))
  for (i in seq_len(nrow(x))) stop_if(portable_match(file.path(root, x$rel[[i]]), x$sha256[[i]], x$bytes[[i]]), paste(name, x$rel[[i]]))
  x
}
require_terms <- function(relative, terms) {
  txt <- tolower(paste(readLines(file.path(root, relative), warn=FALSE, encoding="UTF-8"), collapse=" "))
  for (term in terms) stop_if(grepl(tolower(term), txt, fixed=TRUE), paste("status", relative, term))
}
final_txt <- json_text(file.path(reports, final_name))
for (term in c('"accepted_phase": "13C"', '"baseline": "Infectious Disease System Futures, 2050 / 2075"', '"status": "ACCEPTED / FROZEN"', '"accepted_by": "Sol explicit acceptance decision supplied for this run"', '"accepted_date": "2026-09-10"', paste0('"source_commit": "', source_commit, '"'), '"assumptions": 36', '"transmission_states": 36', '"surveillance_states": 36', '"response_states": 36', '"dependency_states": 36', '"uncertainty_states": 36', '"scenario_sources": 33', '"comparison_rows": 6', '"phase13a_13b_accepted_frozen": true', '"phase1_12_immutable": true', '"phase14_implemented": false', '"numeric_future_values_adopted": false', '"future_case_counts": false', '"future_incidence_forecast": false', '"outbreak_probability": false', '"disease_burden_forecast": false', '"individual_risk_model": false', '"vulnerability_ej_scoring": false', '"unsupported_local_downscaling": false', '"great_black_swamp": "C — HOLD / noncanonical"', '"toledo_intake_coordinate_discrepancy": "UNRESOLVED"')) stop_if(grepl(term, final_txt, fixed=TRUE), paste("final manifest", term))
final_x <- verify_manifest(final_name, length(expected_final), expected_final)
stop_if(!any(final_x$rel == final_name), "final manifest self-list")
working_txt <- json_text(file.path(reports, working_name)); stop_if(grepl('"phase": "13C"', working_txt, fixed=TRUE) && grepl('"status": "implemented_validated_pending_sol_acceptance"', working_txt, fixed=TRUE), "working manifest identity")
working_x <- manifest_entries(file.path(reports, working_name)); stop_if(nrow(working_x)==length(expected_working), "working manifest artifact count")
stop_if(identical(as.character(working_x$rel), expected_working), "working manifest inventory")
for (i in seq_len(nrow(working_x))) stop_if(portable_match(file.path(root, working_x$rel[[i]]), working_x$sha256[[i]], working_x$bytes[[i]]), paste("working artifact", working_x$rel[[i]]))
phase13a_x <- verify_manifest(phase13a_name, 25L); phase13b_x <- verify_manifest(phase13b_name, 24L)
phase13a_txt <- json_text(file.path(reports, phase13a_name)); phase13b_txt <- json_text(file.path(reports, phase13b_name))
stop_if(grepl('"status": "ACCEPTED / FROZEN"', phase13a_txt, fixed=TRUE) && grepl('"status": "ACCEPTED / FROZEN"', phase13b_txt, fixed=TRUE), "prior phase status")
check_txt <- tolower(json_text(file.path(reports, "infectious_disease_scenario_artifact_check.json")))
for (term in c('"status": "passed"', '"assumptions": 36', '"transmission_states": 36', '"surveillance_states": 36', '"response_states": 36', '"dependency_states": 36', '"uncertainty_states": 36', '"scenario_sources": 33', '"comparison_rows": 6', '"phase13a_13b_freeze_integrity": true', '"prior_freeze_manifest_entries_checked": 358', '"prior_unique_protected_artifacts_checked": 351', '"phase13a_artifacts_checked": 25', '"phase13b_artifacts_checked": 24', '"manifest_artifacts_checked": 24', '"active_holds_preserved": true', '"deferred_maintenance_preserved": true', '"phase14_not_implemented": true', '"review_present": true')) stop_if(grepl(term, check_txt, fixed=TRUE), paste("artifact check", term))
require_terms("PROJECT_STATUS.md", c("## phase 13c — accepted / frozen", final_name, "active phase: **none**", "next analytical phase: **not approved**", "phase 14 is not implemented", "great black swamp", "hold", "noncanonical", "intake-coordinate discrepancy", "unresolved"))
require_terms("docs/canon_status.md", c("phase 13c is **accepted / frozen**", final_name, "active phase: **none**", "next analytical phase: **not approved**", "phase 14 is not implemented", "great black swamp", "hold", "noncanonical", "intake-coordinate discrepancy", "unresolved"))
require_terms("reports/current_phase_handoff.md", c("## phase 13c final acceptance / freeze handoff", "phase 13c **accepted / frozen**", final_name, "active phase: **none**", "next analytical phase: **not approved**", "phase 14 is not implemented", "great black swamp", "hold", "noncanonical", "intake-coordinate discrepancy", "unresolved"))
require_terms("README.md", c("phase 13c is **accepted / frozen**", final_name, "phase 13c python freeze validator", "phase 13c independent r freeze validator"))
require_terms("CHANGELOG.md", c("sol formally accepted and froze phase 13c", final_name, "active phase is none", "next analytical phase is not approved"))
require_terms("docs/phase_briefs/phase13c_infectious_disease_futures.md", c("status: approved scope / not implemented", "phase 13c", "2050", "2075", "great black swamp", "toledo intake-coordinate discrepancy"))
review <- tolower(json_text(file.path(reports, "infectious_disease_scenario_independent_review.md")))
for (term in c('"passed": true', '"security_concerns": []', '"logic_errors": []', '"provenance_errors": []', '"epidemiological_errors": []', '"scenario_boundary_errors": []', '"surveillance_errors": []', '"spatial_scale_errors": []', '"health_boundary_errors": []')) stop_if(grepl(term, review, fixed=TRUE), paste("review", term))
prior_names <- c(phase13a_name, phase13b_name, final_name)
prior_files <- list.files(reports, pattern="^phase.*_freeze_manifest[.]json$", full.names=TRUE); prior_files <- prior_files[!basename(prior_files) %in% prior_names]
protected <- character(); expected_hashes <- character(); prior_entries <- 0L
for (path in sort(prior_files)) {
  x <- manifest_entries(path)
  for (i in seq_len(nrow(x))) {
    rel <- x$rel[[i]]; expected <- x$sha256[[i]]
    if (rel %in% names(expected_hashes)) stop_if(unname(expected_hashes[[rel]]) == expected, paste("conflicting prior hash", rel))
    expected_hashes[rel] <- expected; stop_if(portable_match(file.path(root, rel), expected, x$bytes[[i]]), paste("prior hash", rel)); protected <- c(protected, rel); prior_entries <- prior_entries + 1L
  }
}
stop_if(prior_entries == 358L && length(unique(protected)) == 351L, paste("prior inventory", prior_entries, length(unique(protected))))
changed <- system2("git", c("-C", root, "diff", "--name-only", source_commit), stdout=TRUE, stderr=TRUE)
stop_if(length(intersect(changed, unique(c(protected, phase13a_x$rel, phase13b_x$rel)))) == 0L, "protected artifact changed")
for (base in c(file.path(root,"data","processed"), file.path(root,"outputs","maps","systems"), file.path(root,"src","python","systems"), file.path(root,"src","R","systems"))) {
  files <- list.files(base, recursive=TRUE, full.names=TRUE); bad <- files[grepl("phase14|phase_14|(^|[/\\\\])44_", tolower(basename(files)))]
  stop_if(length(bad)==0L, paste("Phase 14 implementation", base))
}
cat(sprintf("Phase 13C R freeze validation passed: %d final artifacts, %d working artifacts; Phase 13A/13B artifacts %d/%d; prior Phase 1–12 inventory %d/%d; review, status surfaces, holds, deferred maintenance, and no Phase 14 implementation verified\n", nrow(final_x), nrow(working_x), nrow(phase13a_x), nrow(phase13b_x), prior_entries, length(unique(protected))))

File: src/python/systems/validate_phase13c_freeze.py
#!/usr/bin/env python3
"""Independent freeze-boundary validation for accepted Phase 13C."""
from __future__ import annotations

import argparse
import hashlib
import json
import re
import subprocess
from pathlib import Path
from typing import Any

EXPECTED_COUNTS = {
    "assumptions": 36,
    "transmission_states": 36,
    "surveillance_states": 36,
    "response_states": 36,
    "dependency_states": 36,
    "uncertainty_states": 36,
    "scenario_sources": 33,
    "comparison_rows": 6,
}
EXPECTED_PRIOR_ENTRIES = 358
EXPECTED_PRIOR_UNIQUE = 351
SOURCE_COMMIT = "c746e361e052e979227e40140c6d923557911524"
FINAL_MANIFEST = "reports/phase13c_infectious_disease_futures_freeze_manifest.json"
WORKING_MANIFEST = "reports/infectious_disease_scenario_manifest.json"
ARTIFACT_CHECK = "reports/infectious_disease_scenario_artifact_check.json"
PHASE13A_MANIFEST = "reports/phase13a_infectious_disease_freeze_manifest.json"
PHASE13B_MANIFEST = "reports/phase13b_infectious_disease_dependencies_freeze_manifest.json"
REVIEW = "reports/infectious_disease_scenario_independent_review.md"
TEXT_SUFFIXES = {".csv", ".json", ".md", ".py", ".r", ".R", ".txt", ".yaml", ".yml", ".toml", ".svg"}
REVIEWS = (REVIEW,)
WORKING_ARTIFACTS = (
    "data/processed/scenarios/infectious_disease_scenario_assumptions.csv",
    "data/processed/scenarios/infectious_disease_transmission_states_scenario.csv",
    "data/processed/scenarios/infectious_disease_surveillance_states_scenario.csv",
    "data/processed/scenarios/infectious_disease_response_states_scenario.csv",
    "data/processed/scenarios/infectious_disease_dependency_states_scenario.csv",
    "data/processed/scenarios/infectious_disease_uncertainty_states_scenario.csv",
    "data/processed/analysis/infectious_disease_scenario_sources.csv",
    "outputs/figures/infectious_disease_scenario_comparison.csv",
    "outputs/maps/systems/43_infectious_disease_system_futures_2050.png",
    "outputs/maps/systems/43_infectious_disease_system_futures_2050.svg",
    "outputs/maps/systems/43b_infectious_disease_system_futures_2075.png",
    "outputs/maps/systems/43b_infectious_disease_system_futures_2075.svg",
    "outputs/figures/infectious_disease_scenario_comparison.png",
    "outputs/figures/infectious_disease_scenario_comparison.svg",
    "reports/infectious_disease_scenario_sources.md",
    "reports/infectious_disease_scenario_assumptions.md",
    "reports/infectious_disease_scenario_findings.md",
    "reports/infectious_disease_scenario_qa.md",
    "reports/phase13c_citation_ledger.json",
    "src/python/systems/build_infectious_disease_futures.py",
    "src/python/systems/validate_infectious_disease_futures.py",
    "src/R/systems/validate_infectious_disease_futures.R",
    ARTIFACT_CHECK,
    REVIEW,
)
EXPECTED_FINAL_ARTIFACTS = WORKING_ARTIFACTS + (
    WORKING_MANIFEST,
    "docs/phase_briefs/phase13c_infectious_disease_futures.md",
    "src/python/systems/validate_phase13c_freeze.py",
    "src/R/systems/validate_phase13c_freeze.R",
)


def load_json(root: Path, relative: str) -> dict[str, Any]:
    with (root / relative).open("r", encoding="utf-8") as handle:
        return json.load(handle)


def artifact_entries(payload: dict[str, Any]) -> dict[str, Any]:
    return payload.get("artifacts", payload.get("files", {}))


def canonical(data: bytes) -> bytes:
    return data.replace(b"\r\n", b"\n").replace(b"\r", b"\n")


def sha256(data: bytes) -> str:
    return hashlib.sha256(data).hexdigest()


def verify_hash(root: Path, relative: str, expected: dict[str, Any] | str) -> str:
    path = root / relative
    assert path.is_file() and path.stat().st_size > 0, f"missing protected artifact: {relative}"
    expected_hash = expected if isinstance(expected, str) else str(expected["sha256"])
    expected_bytes = None if isinstance(expected, str) else expected.get("bytes")
    raw = path.read_bytes()
    candidates = [(sha256(raw), len(raw), "exact")]
    if path.suffix in TEXT_SUFFIXES:
        normalized = canonical(raw)
        candidates.extend(
            [
                (sha256(normalized), len(normalized), "newline-normalized"),
                (sha256(normalized.replace(b"\n", b"\r\n")), len(normalized.replace(b"\n", b"\r\n")), "newline-normalized"),
            ]
        )
    for digest, size, mode in candidates:
        if digest == expected_hash and (expected_bytes is None or int(expected_bytes) == size):
            return mode
    raise AssertionError(f"hash/byte mismatch: {relative}")


def verify_artifacts(root: Path, payload: dict[str, Any], label: str, expected_count: int | None = None) -> dict[str, int]:
    artifacts = artifact_entries(payload)
    assert artifacts, f"{label}: no artifacts"
    if expected_count is not None:
        assert len(artifacts) == expected_count, f"{label}: artifact count {len(artifacts)}"
    exact = 0
    normalized = 0
    for relative, expected in artifacts.items():
        mode = verify_hash(root, relative, expected)
        if mode == "exact":
            exact += 1
        else:
            normalized += 1
    return {"entries": len(artifacts), "exact": exact, "newline_normalized": normalized}


def verify_prior_system(root: Path) -> dict[str, int]:
    excluded = {Path(FINAL_MANIFEST).name, Path(PHASE13A_MANIFEST).name, Path(PHASE13B_MANIFEST).name}
    expected_hashes: dict[str, str] = {}
    protected: set[str] = set()
    entries = 0
    manifests = []
    for path in sorted((root / "reports").glob("phase*_freeze_manifest.json")):
        if path.name in excluded:
            continue
        manifests.append(path)
        payload = json.loads(path.read_text(encoding="utf-8"))
        for relative, metadata in artifact_entries(payload).items():
            expected = str(metadata["sha256"] if isinstance(metadata, dict) else metadata)
            if relative in expected_hashes:
                assert expected_hashes[relative] == expected, f"conflicting prior hash: {relative}"
            expected_hashes[relative] = expected
            verify_hash(root, relative, expected)
            protected.add(relative)
            entries += 1
    assert entries == EXPECTED_PRIOR_ENTRIES, entries
    assert len(protected) == EXPECTED_PRIOR_UNIQUE, len(protected)
    changed = set(subprocess.check_output(["git", "-C", str(root), "diff", "--name-only", SOURCE_COMMIT], text=True).splitlines())
    assert not changed.intersection(protected), sorted(changed.intersection(protected))
    return {"manifest_count": len(manifests), "entries": entries, "unique": len(protected)}


def verify_review(root: Path) -> dict[str, bool]:
    for relative in REVIEWS:
        assert (root / relative).is_file() and (root / relative).stat().st_size > 0, relative
    text = (root / REVIEW).read_text(encoding="utf-8")
    match = re.search(r"```json\s*(\{.*?\})\s*```", text, flags=re.S)
    assert match, "final independent-review JSON block missing"
    verdict = json.loads(match.group(1))
    assert verdict.get("passed") is True
    for key in (
        "security_concerns", "logic_errors", "provenance_errors", "epidemiological_errors",
        "scenario_boundary_errors", "surveillance_errors", "spatial_scale_errors", "health_boundary_errors",
    ):
        assert verdict.get(key) == [], key
    return {"passed": True, "blocking_arrays_empty": True, "review_preserved": True}


def verify_working_package(root: Path) -> dict[str, int]:
    working = load_json(root, WORKING_MANIFEST)
    assert working["phase"] == "13C"
    assert working["status"] == "implemented_validated_pending_sol_acceptance"
    assert working["counts"] == EXPECTED_COUNTS
    assert working["phase13a_13b_accepted_frozen"] is True
    assert working["phase1_12_immutable"] is True
    assert working["phase14_implemented"] is False
    assert list(artifact_entries(working)) == list(WORKING_ARTIFACTS)
    result = verify_artifacts(root, working, "Phase 13C working manifest", len(WORKING_ARTIFACTS))
    assert working["phase13a_freeze_manifest"]["path"] == PHASE13A_MANIFEST
    assert working["phase13b_freeze_manifest"]["path"] == PHASE13B_MANIFEST
    verify_hash(root, PHASE13A_MANIFEST, working["phase13a_freeze_manifest"])
    verify_hash(root, PHASE13B_MANIFEST, working["phase13b_freeze_manifest"])
    return result


def verify_artifact_check(root: Path) -> None:
    check = load_json(root, ARTIFACT_CHECK)
    assert check["status"] == "passed"
    for key, expected in EXPECTED_COUNTS.items():
        assert check[key] == expected, key
    assert check["phase13a_13b_freeze_integrity"] is True
    assert check["prior_freeze_manifest_entries_checked"] == EXPECTED_PRIOR_ENTRIES
    assert check["prior_unique_protected_artifacts_checked"] == EXPECTED_PRIOR_UNIQUE
    assert check["phase13a_artifacts_checked"] == 25
    assert check["phase13b_artifacts_checked"] == 24
    assert check["manifest_artifacts_checked"] == len(WORKING_ARTIFACTS)
    assert check["active_holds_preserved"] is True
    assert check["deferred_maintenance_preserved"] is True
    assert check["phase14_not_implemented"] is True
    assert check["review_present"] is True


def verify_no_phase14(root: Path) -> None:
    for base in (root / "data/processed", root / "outputs/maps/systems", root / "src/python/systems", root / "src/R/systems"):
        for path in base.rglob("*"):
            if path.is_file():
                name = path.name.lower()
                assert "phase14" not in name and "phase_14" not in name, str(path)
                assert not name.startswith(("44_",)), str(path)


def require_terms(root: Path, relative: str, terms: tuple[str, ...]) -> None:
    text = (root / relative).read_text(encoding="utf-8").lower()
    for term in terms:
        assert term in text, (relative, term)


def verify_status_surfaces(root: Path) -> None:
    require_terms(root, "PROJECT_STATUS.md", (
        "## phase 13c — accepted / frozen", FINAL_MANIFEST.lower(),
        "active phase: **none**", "next analytical phase: **not approved**",
        "phase 14 is not implemented", "great black swamp", "hold", "noncanonical",
        "intake-coordinate discrepancy", "unresolved",
    ))
    require_terms(root, "docs/canon_status.md", (
        "phase 13c is **accepted / frozen**", FINAL_MANIFEST.lower(),
        "active phase: **none**", "next analytical phase: **not approved**",
        "phase 14 is not implemented", "great black swamp", "hold", "noncanonical",
        "intake-coordinate discrepancy", "unresolved",
    ))
    require_terms(root, "reports/current_phase_handoff.md", (
        "## phase 13c final acceptance / freeze handoff", "phase 13c **accepted / frozen**",
        FINAL_MANIFEST.lower(), "active phase: **none**", "next analytical phase: **not approved**",
        "phase 14 is not implemented", "great black swamp", "hold", "noncanonical",
        "intake-coordinate discrepancy", "unresolved",
    ))
    require_terms(root, "README.md", (
        "phase 13c is **accepted / frozen**", FINAL_MANIFEST.lower(),
        "phase 13c python freeze validator", "phase 13c independent r freeze validator",
    ))
    require_terms(root, "CHANGELOG.md", (
        "sol formally accepted and froze phase 13c", FINAL_MANIFEST.lower(),
        "active phase is none", "next analytical phase is not approved",
    ))
    require_terms(root, "docs/phase_briefs/phase13c_infectious_disease_futures.md", (
        "status: approved scope / not implemented", "phase 13c", "2050", "2075",
        "great black swamp", "toledo intake-coordinate discrepancy",
    ))


def main() -> None:
    parser = argparse.ArgumentParser()
    parser.add_argument("root", nargs="?", default=None)
    args = parser.parse_args()
    root = Path(args.root).resolve() if args.root else Path(__file__).resolve().parents[3]

    final = load_json(root, FINAL_MANIFEST)
    assert final["accepted_phase"] == "13C"
    assert final["baseline"] == "Infectious Disease System Futures, 2050 / 2075"
    assert final["status"] == "ACCEPTED / FROZEN"
    assert final["accepted_by"] == "Sol explicit acceptance decision supplied for this run"
    assert final["accepted_date"] == "2026-09-10"
    assert final["source_commit"] == SOURCE_COMMIT
    assert final["counts"] == EXPECTED_COUNTS
    assert final["phase13a_freeze_manifest"]["path"] == PHASE13A_MANIFEST
    assert final["phase13b_freeze_manifest"]["path"] == PHASE13B_MANIFEST
    assert final["protected_inputs"] == [PHASE13A_MANIFEST, PHASE13B_MANIFEST, WORKING_MANIFEST]
    assert final["phase13a_13b_accepted_frozen"] is True
    assert final["phase1_12_immutable"] is True
    assert final["phase14_implemented"] is False
    for key in ("numeric_future_values_adopted", "future_case_counts", "future_incidence_forecast", "outbreak_probability", "disease_burden_forecast", "individual_risk_model", "vulnerability_ej_scoring", "unsupported_local_downscaling"):
        assert final[key] is False, key
    assert final["active_holds_preserved"] == {
        "great_black_swamp": "C — HOLD / noncanonical",
        "toledo_intake_coordinate_discrepancy": "UNRESOLVED",
    }
    assert final["deferred_maintenance_preserved"] == [
        "Phase 6B manifest status wording mismatch",
        "Phase 3A missing manifest status",
    ]
    assert list(artifact_entries(final)) == list(EXPECTED_FINAL_ARTIFACTS)
    assert FINAL_MANIFEST not in artifact_entries(final)
    final_result = verify_artifacts(root, final, "Phase 13C final freeze manifest", len(EXPECTED_FINAL_ARTIFACTS))

    working_result = verify_working_package(root)
    verify_artifact_check(root)
    phase13a = load_json(root, PHASE13A_MANIFEST)
    phase13b = load_json(root, PHASE13B_MANIFEST)
    assert phase13a["status"] == "ACCEPTED / FROZEN"
    assert phase13b["status"] == "ACCEPTED / FROZEN"
    phase13a_result = verify_artifacts(root, phase13a, "Phase 13A final freeze manifest", 25)
    phase13b_result = verify_artifacts(root, phase13b, "Phase 13B final freeze manifest", 24)
    prior = verify_prior_system(root)
    review = verify_review(root)
    verify_no_phase14(root)
    verify_status_surfaces(root)

    result = {
        "status": "passed",
        "phase": "13C",
        "freeze_manifest": {"path": FINAL_MANIFEST, **final_result},
        "working_manifest": working_result,
        "phase13a_integrity": {"final_artifacts": phase13a_result["entries"]},
        "phase13b_integrity": {"final_artifacts": phase13b_result["entries"]},
        "prior_system_immutability": prior,
        "review_history": review,
        "counts": EXPECTED_COUNTS,
        "active_holds_preserved": True,
        "deferred_maintenance_preserved": True,
        "phase14_not_implemented": True,
    }
    print(json.dumps(result, indent=2, ensure_ascii=False))


if __name__ == "__main__":
    main()
